lib/facility_factory.rb

```diff
@@ -0,0 +1,65 @@
+class FacilityFactory
+
+  def initialize
+  end
+
+  def create_facilities(state_abbv)
+    if state_abbv == "CO"
+      create_co_facilities
+      elsif state_abbv == "MO"
+        create_mo_facilities
+          elsif state_abbv == "NY"
+            create_ny_facilities
+              else
+                "sorry, we don't yet serve #{state_abbv}"
+              end
+  end
+
+  def create_co_facilities
+    facility_data = DmvDataService.new.co_dmv_office_locations
+    facilities = []
+    facility_data.each do |facility|
+      facility_data = {
+        name:facility[:dmv_office],
+        address:"#{facility[:address_li]}, #{facility[:address__1]}, #{facility[:city]}, #{facility[:state]}, #{facility[:zip]}",
+        phone:facility[:phone],
+        services:facility[:services_p],
+        collected_fees:0,
+        registered_vehicles:[]}
+      facilities << Facility.new(facility_data)
+    end
+    facilities
+  end
+
+  def create_ny_facilities
+    facility_data = DmvDataService.new.ny_dmv_office_locations
+    facilities = []
+    facility_data.each do |facility|
+      facility_data = {
+        name:facility[:office_name],
+        address:"#{facility[:street_address_line_1]}, #{facility[:city]}, #{facility[:state]}, #{facility[:zip_code]}",
+        phone:facility[:public_phone_number],
+        services:[],
+        collected_fees:0,
+        registered_vehicles:[]}
+      facilities << Facility.new(facility_data)
+    end
+    facilities
+  end
+
+  def create_mo_facilities
+    facility_data = DmvDataService.new.mo_dmv_office_locations
+    facilities = []
+    facility_data.each do |facility|
+      facility_data = {
+        name:facility[:name],
+        address:"#{facility[:address1]}, #{facility[:city]}, #{facility[:state]}, #{facility[:zipcode]}",
+        phone:facility[:phone],
+        services:[],
+        collected_fees:0,
+        registered_vehicles:[]}
+      facilities << Facility.new(facility_data)
+    end
+    facilities
+  end
```

Comment:
In your submission form you asked for feedback on these methods (bringing in data from other states) and suggestions on how to make them leaner. A few ideas/foods for thought:
- is there an enumerable that would eliminate the need for lines 52 and 63?
- I don't feel this is a top priority refactor, but have you considered abstracting the contents of the block to an additional helper method? I'd imagine that way this method might look something like
```
def create_mo_facilities
  facility_datas = DmvDataService.new.mo_dmv_office_locations
  facility_datas.map do |facility_data|
    mo_facility(facility_data)
  end
end
```
That way the helper method could have a single responsibility: making a facility instance from a subset of the MO data.
```
def mo_facility(data)
  Facility.new({
    name: data[:name],
    ...
  })
end
```